src/main/resources/META-INF/plugin.xml

```diff
@@ -6,12 +6,15 @@
     <depends>com.intellij.modules.lang</depends>
     <depends>com.intellij.modules.platform</depends>
     <!--
+
       Note that since 2022.3:
       * the plugin *folder* under $PRODUCT_ROOT$/plugins/ is called `vcs-git` and not `git4idea`, but
       * the <id> of the plugin, as recorded in plugin's META-INF/plugin.xml is still `Git4Idea` - and that's what's expected here in `<depends>`
     -->
     <depends>Git4Idea</depends>
 
+    <depends>org.jetbrains.plugins.github</depends>
```

Comment:
Add as optional dependency (as this plugin might to disabled). Detect in our plugin code whether it's available (just `try { ... } catch (ClassNotFound ...) { ... }` is probably a valid approach).

Reply:
(ofc assuming we stick to the current approach... subject to whether we can live with the caveats 1. and 2. from PR discussion)